File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number):
    if number <= 0:
        raise ValueError("Only positive integers are allowed")
    else:
        return stepsAux(number)
        
def stepsAux(number):
    if number == 1:
        return 0
    else:
        if number % 2 == 0:
            return 1 + stepsAux(number / 2)
        else:
            return 1 + stepsAux(3 * number + 1)

File: solutions/javascript/collatz-conjecture/1/collatz-conjecture.js
//
// This is only a SKELETON file for the 'Collatz Conjecture' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const steps = (n) => {
  if(n <= 0)
    throw new Error('Only positive numbers are allowed');
  let counter = 0
  while(true) {
    if(n === 1)
      break;
    counter++;
    if(n % 2 === 0)
      n /= 2;
    else
      n = 3 * n + 1;
  }
  return counter;
};
